UndertaleModTool/Converters/UndertaleCachedImageLoader.cs

```diff
@@ -426,19 +416,18 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
                             {
                                 for (int j = 0; j < w * depth; j += depth)
                                 {
-                                    int origIndex = (y1 * data.Stride) + (i * data.Stride) + (x1 * depth) + j;
+                                    int origIndex = (y1 * tilePageBMP.RowBytes) + (i * tilePageBMP.RowBytes) + (x1 * depth) + j;
                                     int croppedIndex = (i * w * depth) + j;
 
                                     Buffer.BlockCopy(buffer, origIndex, bufferRes, croppedIndex, depth);
                                 }
                             }
                         }
 
-                        Bitmap tileBMP = new(w, h);
-                        BitmapData tileData = tileBMP.LockBits(new Rectangle(0, 0, w, h), ImageLockMode.WriteOnly, format);
-                        Marshal.Copy(bufferRes, 0, tileData.Scan0, bufferResLen);
-                        tileBMP.UnlockBits(tileData);
+                        SKBitmap tileBMP = new(w, h);
+                        Marshal.Copy(bufferRes, 0, tileBMP.GetPixels(), bufferResLen);
                         ArrayPool<byte>.Shared.Return(bufferRes);
+                        tileBMP.SetImmutable();
```

Comment:
same here

Reply:
But I'm sure that it will have a better performance here, at least because of this: https://github.com/mono/SkiaSharp/issues/2112#issuecomment-1214726409
`tileSurf.Canvas.DrawBitmap(srcBMP, 0, 0)` and `layerG.DrawBitmap(TileCache[new(tilesBG.Texture.Name.Content, id)], x * w, y * h);`